### PR TITLE
Refactor some of the repo for an external audience

### PR DESCRIPTION
**Pull request by paultyng**
The goal of this PR is to focus this repo on external provider developers. Mostly it involves removing things specific to our workflows (ie. scripts, website, vendoring, etc) that are no longer relevant with Go updates (module proxy) and registry provider support. This goes along with #9.

TODO: 

* [ ] Add GitHub action for go build / gofmt / go unit test?

### CODE REVIEW

**Comment by justincampbell** [comment]
Would it be possible to remove the vendor dir in a separate PR? That would make it easier for me to understant the other changes.

**Comment by paultyng** [comment]
@justincampbell I moved it all to a single commit, so you can view by commit and skip it.

File: README.md [hunk 1]
```diff
@@ -16,42 +15,33 @@ Please see the [GitHub template repository documentation](https://help.github.co
 Requirements
 ------------
 
--	[Terraform](https://www.terraform.io/downloads.html) 0.10.x
--	[Go](https://golang.org/doc/install) 1.11
+-	[Terraform](https://www.terraform.io/downloads.html) 0.12.x
+-	[Go](https://golang.org/doc/install) 1.12
```
Comment (appilon) [inline README.md]: we could recommend 1.14

Comment (appilon) [reply README.md]: I guess TF 0.12 is built on 1.12... not sure if there is any real risk of a 1.12 built TF talking to a 1.14 built provider

File: README.md [hunk 2]
```diff
@@ -62,22 +52,9 @@ Fill this in for each provider
 Developing the Provider
 ---------------------------
 
-If you wish to work on the provider, you'll first need [Go](http://www.golang.org) installed on your machine (version 1.8+ is *required*). You'll also need to correctly setup a [GOPATH](http://golang.org/doc/code.html#GOPATH), as well as adding `$GOPATH/bin` to your `$PATH`.
-
-To compile the provider, run `make build`. This will build the provider and put the provider binary in the `$GOPATH/bin` directory.
+If you wish to work on the provider, you'll first need [Go](http://www.golang.org) installed on your machine (version 1.12+ is *required*).
```
Comment (appilon) [inline README.md]: if we bump to 1.14 reflect it here

**Review by appilon — APPROVED** [review]
I'm happy with making this a more external/community driven template, just keep in mind this repo is used as a github template for the dev program (I think?) so maybe replicate it in another repo? Or alternatively make a new repo called `terraform-provider-example` to serve the community example? Either way, just wanna make sure we don't lose something we still need. Otherwise love the simplicity!

File: README.md (hunk 1 above)
Comment (paultyng) [reply README.md]: 1.12 isn't even listed anymore, so yeah we should just bump to 1.14

Comment (paultyng) [reply README.md]: https://golang.org/dl/

Comment (kmoe) [reply README.md]: We could recommend 1.14 as it's the current Go version, but it's not _required_ for provider development. Provider developers may be working in environments where they are unable to upgrade the Go version. From the SDK point of view, Go 1.12+ is required.

**Comment by paultyng** [comment]
I clarified the language to say `>= 1.12` as well as removed the second mention to simplify the maintenance a bit.

File: GNUmakefile [hunk 3]
```diff
@@ -1,74 +1,5 @@
...
-website-test:
-ifeq (,$(wildcard $(GOPATH)/src/$(WEBSITE_REPO)))
-	echo "$(WEBSITE_REPO) not found in your GOPATH (necessary for layouts and assets), get-ting..."
-	git clone https://$(WEBSITE_REPO) $(GOPATH)/src/$(WEBSITE_REPO)
-endif
-	@$(MAKE) -C $(GOPATH)/src/$(WEBSITE_REPO) website-provider-test PROVIDER_PATH=$(shell pwd) PROVIDER_NAME=$(PKG_NAME)
-
-.PHONY: build test testacc testrace cover vet fmt fmtcheck errcheck test-compile website website-test
+# Run acceptance tests
+testacc:
```
Comment (justincampbell) [inline GNUmakefile:5]:
```suggestion
.PHONY: testacc
testacc:
```

**Review by justincampbell — APPROVED** [review]

**Pull request merged**